features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp

```diff
@@ -168,17 +169,16 @@ nsapi_size_or_error_t UBLOX_AT_CellularStack::socket_sendto_impl(CellularSocket
     MBED_ASSERT(socket->id != -1);
 
     int sent_len = 0;
-    pollfh fhs;
-    fhs.fh = _at.get_file_handle();
-    fhs.events = POLLIN;
+    uint8_t ch = 0;
 
     if (socket->proto == NSAPI_UDP) {
         if (size > UBLOX_MAX_PACKET_SIZE) {
             return NSAPI_ERROR_PARAMETER;
         }
         _at.cmd_start_stop("+USOST", "=", "%d%s%d%d", socket->id, address.get_ip_address(), address.get_port(), size);
 
-        (void)poll(&fhs, 1, 50);
+        rtos::ThisThread::sleep_for(50);
```

Comment:
can't we keep (void)poll(&fhs, 1, 50); instead sleep_for ?

Reply:
polls logic didnt work.

Reply:
It is in riotMicro board as well. If it didn't work then it should be replaced.